lib/application.js

```diff
@@ -383,6 +384,50 @@ app.set = function set(setting, val) {
   return this;
 };
 
+/**
+ * Extend `setting` with `val`, or return `setting`'s value.
+ *
+ *    app.set('foo', { 'bar' : 1 })
+ *    app.expand('foo', { 'baz' : 1 });
+ *    app.expand('foo');
+ *    // => { "bar" : 1, "baz" : 1 }
+ *
+ * Mounted servers inherit their parent server's settings.
+ *
+ * @param {String} setting
+ * @param {*} [val]
+ * @return {Server} for chaining
+ * @public
+ */
+
+app.extend = function(setting, val) {
+  if (arguments.length === 1) {
+    // app.get(setting)
+    return this.settings[setting];
+  }
+
+  if (typeof val === 'undefined') {
+    return this;
+  } else if (val === null || typeof val !== 'object' || val instanceof Array) {
```

Comment:
I would recommend `Array.isArray` instead of `instanceof Array`. All the `instanceof Array`s were removed in Express 1.0.

Reply:
When looking at node.green, I think in my other comment I mentioned that I thought that wasn't supported in node `0.12`, but I think I looked at the wrong field, and it has been so long since I have used `0.12` I couldn't remember what was supported...

http://node.green/#ES2015-subclassing-Array-is-subclassable-Array-isArray-support